app/services/sign_in/token_exchanger.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module SignIn
+  class TokenExchanger
+    include ActiveModel::Validations
```

Comment:
✂️ I don't think you need this anymore with your changes.

Reply:
done